### PR TITLE
update links and formatting, project landing pages

### LINKED ISSUE #12661: [beta] project landing pages need CDN link and link to test suite pen
The following files need to be updated with the correct URL for the blank pen which houses the test suites (right now it links to the same pen on an un-official codepen account which I was using to test these) and the correct CDN link (it currently reflects my repo, but have a new CDN since the repo transferred ownership):

`freeCodeCamp/seed/challenges/01-responsive-web-design/responsive-web-design-projects.json`
`freeCodeCamp/seed/challenges/03-front-end-libraries/front-end-libraries-projects.json`
`freeCodeCamp/seed/challenges/04-data-visualization/data-visualization-projects.json`

I'm happy to take this one on since I know what needs to be changed where.

Also @QuincyLarson - 1 thing:
- I know we were trying not to have a lot of copy for these, but I think for the calc project, the landing page is a good place to make a note about the 2 different types of calc logic, since this issue has been raised so many times. (including one just today :smile: ) It is a bit long to include in the user stories themselves. 

Here is the note:

> Note On Calculator Logic: It should be noted that there are two main schools of thought on calculator input logic: immediate execution and formula logic (our example utilizes formula logic and observes order of operation precedence, immediate execution does not). Either is acceptable, but please note that depending on which you choose, your calculator may yield different results than ours for certain equations (see below example). As long as your math can be verified by another production calculator, please do not consider this a bug.
>
>EXAMPLE: 
> - 3 + 5 x 6 - 2 / 4 =
    - Immediate Execution Logic: 11.5
    - Formula/Expression Logic: 32.5

### PR DESCRIPTION
**Pull request by no-stack-dub-sack**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] curriculum update
- [ ] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue: Closes #12661

#### Description

@QuincyLarson  this PR addresses #12661:
- updates all project landing pages with correct test suite CDN link
- updates all project landing pages with correct test suite code pen links
- added Tree Map D3 project 
- added dataset links for all D3 projects

There's still a bit to be done with the landing pages, so we will need to open a separate issue for that and/or condense already existing issues

### CODE REVIEW

File: seed/challenges/03-front-end-libraries/front-end-libraries-projects.json
Comment (raisedadead) [inline seed/challenges/03-front-end-libraries/front-end-libraries-projects.json]: Break this into multiple lines:

image

File: seed/challenges/04-data-visualization/data-visualization-projects.json
Comment (raisedadead) [inline seed/challenges/04-data-visualization/data-visualization-projects.json]: Inconsistent bullet points

image

**Review by raisedadead — CHANGES_REQUESTED** [review]

**Comment by raisedadead** [comment]
Do we really need bullet points, I am not a huge fan but this could just simply follow the same format as that of the previous JS and HTML challenges?

We already have less real estate, with the large font size. What do you think?

**Comment by no-stack-dub-sack** [comment]
@raisedadead I'm ok with going no bullet points, I only did it there for consistency with the D3 projects that have multiple datasets which I felt needed them: 
image

But I guess I could just leave them there where they make sense, and get rid of them everywhere else. Sound good? Or do you think I should just ditch them there too? I thought, for that one and the other with multiple, that it was important to let the user know what the particular dataset was for

**Comment by raisedadead** [comment]
Yeah, I think the label on top of the resources is important. And I also agree that you should leave the bullet points where they make sense as in the example above.

**Comment by no-stack-dub-sack** [comment]
@raisedadead ok, so I'll change it to be in only those challenges that have multiple datasets, and leave the rest without bullet points. I still think it looks a little funny, be was have bigger things to worry about for the time being so this will do for now.

**Comment by raisedadead** [comment]
I think this needs a styling overhaul. but yeah if you could address the first comment about breaking the instructions to at least multiline, we can ignore the other aesthetics for now.

**Comment by no-stack-dub-sack** [comment]
@raisedadead about to make a PR and I think this does it for now - added the multiline for the example like you said, and removed the bullets where they were not needed (from every challenge except D3 tree map and D3 choropleth where multiple datasets are needed)

Also, I definitely agree with you that these sections need an overall style overhaul 😄

**Comment by no-stack-dub-sack** [comment]
@raisedadead WAIT! Don't merge this yet - looks like #12720 affects this entire section, so once that PR is merged, I will rebase off of that before this gets merged. Does that make sense or should the merge still go ok if this is merged first?

**Comment by raisedadead** [comment]
Blocked for merge.

**Comment by raisedadead** [comment]
@no-stack-dub-sack I have resolved a merge conflict with a PR that came in with another PR https://github.com/freeCodeCamp/freeCodeCamp/pull/12909

Can you please do a sanity check and let me know, for a merge?

Also we need not wait for #12720, it already has conflicts and should be resolved separately.

**Review by raisedadead — APPROVED** [review]

**Comment by no-stack-dub-sack** [comment]
@raisedadead Yup, if all you changed is adding the closing `</a>` tags to resolve that conflict than LGTM!

**Pull request merged**
